R/zchunk_L142.pfc_R_S_T_Y.R

```diff
@@ -8,83 +8,179 @@
+      mutate(Non.CO2 = "C2F6") ->
+      EDGAR_C2F6
+
+    EDGAR_CF4 %>%
+      mutate(Non.CO2 = "CF4") %>%
+      bind_rows(EDGAR_SF6, EDGAR_C2F6,.) ->
+      ALL_EDGAR_HFC
+
+    Other_F %>%
+      rename(Non.CO2 = Gas) ->
+      Other_F
+
+    # Then, prepare EDGAR data for use.
+    # Map in region ID and sector name, and remove year 1970 to match "HISTORICAL_YEARS" constant.
+
+    ALL_EDGAR_HFC %>%
+      left_join_error_no_match(EDGAR_sector, by = "IPCC_description") %>%
+      standardize_iso(col = "ISO_A3") %>%
+      change_iso_code('rou','rom') %>%                                        # Switch Romania iso code to its pre-2002 value
+      left_join_error_no_match(iso_GCAM_regID, by = "iso")  %>%
+      rename(EDGAR_agg_sector = agg_sector) %>%
+      select(GCAM_region_ID, EDGAR_agg_sector, Non.CO2, year, value)  %>%
+      filter(year !="1970") ->
+      L142.EDGAR_HFC
+
+    L142.EDGAR_HFC %>%
+      group_by(GCAM_region_ID, EDGAR_agg_sector, Non.CO2, year) %>%
+      summarize(value = sum(value))  %>%
+      rename(EDGAR_emissions = value) ->
+      L142.EDGAR_PFC_R_S_T_Yh.tmp1
+
+    # Map in other f-gas sector, which varies by gas.
+    # Seperate PFCs and add non-CO2 gas information, then recombine.
+
+    L142.EDGAR_PFC_R_S_T_Yh.tmp1 %>%
+      filter(EDGAR_agg_sector != "other_f_gases") ->
+      L142.EDGAR_PFC_R_S_T_Yh_rest
+
+    L142.EDGAR_PFC_R_S_T_Yh.tmp1 %>%
+      filter(EDGAR_agg_sector == "other_f_gases")  %>%
+      ungroup() %>%
+      left_join(Other_F, by = "Non.CO2")   %>%
+      select(-EDGAR_agg_sector)  %>%
+      rename(EDGAR_agg_sector = Sector)  %>%
+      bind_rows(L142.EDGAR_PFC_R_S_T_Yh_rest,.)  %>%
+      ungroup() ->                                              # ungroup needed for later `left_join_keep_first_only`
+      L142.EDGAR_PFC_R_S_T_Yh.tmp1
+
+    # Map PFC gases to GCAM technologies.
+
+    GCAM_tech %>%
+      repeat_add_columns(tibble(GCAM_region_ID = GCAM_region_names$GCAM_region_ID)) %>%
+      repeat_add_columns(tibble(year = emissions.EDGAR_YEARS)) %>%
+      repeat_add_columns(tibble("Non.CO2" = emissions.PFCS))  %>%
+      left_join_keep_first_only(L142.EDGAR_PFC_R_S_T_Yh.tmp1, by = c("GCAM_region_ID", "year", "EDGAR_agg_sector", "Non.CO2")) %>%
+      select(GCAM_region_ID, supplysector, subsector, stub.technology, Non.CO2, year, EDGAR_emissions)  %>%
+      replace_na(list(EDGAR_emissions = 0)) ->
+      L142.pfc_R_S_T_Yh.tmp1
+
+    # Disaggregate cooling emissions to residential and commercial sectors.
+
+    L144.in_EJ_R_bld_serv_F_Yh %>%
+      filter(service %in% c("comm cooling", "resid cooling") & fuel == "electricity") ->
+      L142.R_cooling_T_Yh
+
+    L142.R_cooling_T_Yh %>%
+      group_by(GCAM_region_ID, year)  %>%
+      summarize(value = sum(value)) ->
+      L142.R_cooling_Yh
+
+    # Calculate emission shares for residential and commerical sectors.
+
+    L142.R_cooling_Yh %>%
+      left_join(L142.R_cooling_T_Yh, by = c("GCAM_region_ID", "year"))  %>%
```

Comment:
NO!  The 2 files are not the same length, and the extra rows are needed for later.  I'll make a comment in the file about this.